lib/credentials_from_file.c

```diff
@@ -41,46 +41,74 @@ bool write_normal_device_credentials(const char *dev_cred_file,
 		return false;
 	}
 #ifndef NO_PERSISTENT_STORAGE
-	sdow_t sdowriter, *sdow = &sdowriter;
 
-	if (!sdow_init(sdow)) {
+	sdow_t *sdow = sdo_alloc(sizeof(sdow_t));
+	if (!sdow_init(sdow) || !sdo_block_alloc(&sdow->b) || !sdow_encoder_init(sdow)) {
```

Comment:
There are multiple checks here, so the error log should also be updated. Otherwise, there should be separate checks and corresponding error logs.

Is there a possibility to include sdo_block_alloc and sdow_encoder_init inside sdow_init?

Considering sdow_init() calls sdo_block_reset, shouldn't sdo_block_alloc() be called always after sdow_init?

Reply:
The methods are independent to make it symmetrical to 'sdor_*' operations, wherein, the block allocation and the parser initialization needs to be done separately from sdor_init(). Additionally, they are separate operations in both sdor and sdow use-cases. The usage of these methods will be updated in the documentation of sdoblockio's methods in a future PR.